util/db/db.go

```diff
@@ -85,6 +85,8 @@ type ArgoDB interface {
 	AddGPGPublicKey(ctx context.Context, keyData string) (map[string]*appv1.GnuPGPublicKey, []string, error)
 	// DeleteGPGPublicKey removes a GPG public key from the configuration
 	DeleteGPGPublicKey(ctx context.Context, keyID string) error
+	// LogInClusterWarning checks the in-cluster configuration and prints out any warnings.
+	LogInClusterWarning()
```

Comment:
Pedantic, but: maybe instead of changing the interface, move the function to this package, make it private, and have it just accept `db` as a param instead of making it a receiver function.

Reply:
I tried something like that but there were some complaints from linter that I couldn't resolve.

Reply:
How about this? https://github.com/terrytangyuan/argo-cd/pull/1

Basically, move the responsibility into the server code and leave `db` out of it entirely. It's a bit of duplication, but nicely separates concerns.